includes/fbutils.php

```diff
@@ -42,6 +42,90 @@ class WC_Facebookcommerce_Utils {
 			'female' => 1,
 			'unisex' => 1,
 		);
+
+		/**
+		 * A deferred events storage.
+		 *
+		 * @var array
+		 *
+		 * @since x.x.x
+		 */
+		private static $deferred_events = [];
+
+		/**
+		 * Prints deferred events into page header.
+		 *
+		 * @return void
+		 *
+		 * @since x.x.x
+		 */
+		public static function print_deferred_events() {
+			$deferred_events = static::load_deferred_events();
+			if ( ! empty( $deferred_events ) ) {
+				echo '<script>' . implode( PHP_EOL, $deferred_events ) . '</script>'; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped --- Printing hardcoded JS tracking code.
+			}
+		}
+
+		/**
+		 * Loads deferred event from the storage.
+		 *
+		 * @return array
+		 *
+		 * @since x.x.x
+		 */
+		private static function load_deferred_events(): array {
+			$transient_key = static::get_deferred_events_transient_key();
+			if ( $transient_key ) {
+				$deferred_events = get_transient( $transient_key );
+				if ( $deferred_events ) {
+					delete_transient( $transient_key );
+					return $deferred_events;
+				}
+			}
+			return array();
+		}
+
+		/**
+		 * Adds event into the list of events to be saved/rendered.
+		 *
+		 * @param string $code Generated JS code string w/o a script tag.
+		 *
+		 * @return void
+		 *
+		 * @since x.x.x
+		 */
+		private static function add_deferred_event( string $code ): void {
+			static::$deferred_events[] = $code;
+		}
+
+		/**
+		 * Saves deferred events into the storage.
+		 *
+		 * @return void
+		 *
+		 * @since x.x.x
+		 */
+		public static function save_deferred_events() {
```

Comment:
Is there a scenario where we are saving deferred events over other deferred events that have not been printed yet?